src/NUnitFramework/framework/Constraints/Comparers/IComparer.cs

```diff
@@ -0,0 +1,45 @@
+﻿// ***********************************************************************
+// Copyright (c) 2009 Charlie Poole, Rob Prouse
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+
+namespace NUnit.Framework.Constraints.Comparers
+{
+    /// <summary>
+    /// Interface for comparing two <see cref="Object"/>s.
+    /// </summary>
+    internal interface IComparer
```

Comment:
It will become a pain to actually use `System.Collections.IComparer` in all the files inside or which import the `NUnit.Framework.Constraints.Comparers` namespace. The refactor is a matter of renaming the file and saying OK when VS2015 or VS2017 asks if you want t orename the code symbol. @mikkelbu what is your opinion?